app/controllers/coronavirus_form/product_details_controller.rb

```diff
@@ -78,27 +81,61 @@ def validate_missing_fields
     required.each_with_object([]) do |field, invalid_fields|
       next if @product[field.to_sym].present?
 
-      invalid_fields << {
-        field: field,
-        text: t("coronavirus_form.questions.#{controller_name}.#{field}.custom_error",
-                default: t("coronavirus_form.errors.missing_mandatory_text_field",
-                           field: t("coronavirus_form.questions.#{controller_name}.#{field}.label")).humanize),
-      }
+      invalid_fields << error_response(field)
     end
   end
 
+  def validate_product_quantity
+    field = "product_quantity"
+    quantity = Integer(@product[field.to_sym])
+    return if quantity && quantity.positive?
+
+    error_response(field)
+  rescue TypeError, ArgumentError
+    error_response(field)
+  end
+
+  def validate_product_cost
+    field = "product_cost"
+    cost = Float(@product[field.to_sym])
+    return if cost && (cost.positive? || cost.zero?) && @product[field.to_sym].match?(/^\d+\.?\d{0,2}$/)
+
+    error_response(field)
+  rescue TypeError, ArgumentError
+    error_response(field)
+  end
+
+  def validate_lead_time
+    field = "lead_time"
+    days = Integer(@product[field.to_sym])
+    return if days && (days.positive? || days.zero?)
+
+    error_response(field)
+  rescue TypeError, ArgumentError
+    error_response(field)
+  end
+
   def sanitized_product(params)
     {
       product_id: strip_tags(params[:product_id]).presence || SecureRandom.uuid,
       equipment_type: strip_tags(params[:equipment_type]).presence,
       product_name: strip_tags(params[:product_name]).presence,
-      product_quantity: strip_tags(params[:product_quantity]).presence,
-      product_cost: strip_tags(params[:product_cost]).presence,
+      product_quantity: strip_tags(params[:product_quantity]&.gsub(",", "")&.strip).presence,
+      product_cost: strip_tags(params[:product_cost]&.gsub(/[,£a-zA-Z]/, "")&.strip).presence,
```

Comment:
If a user enters "About £4.00 for a box", we'll use `4.00`. I wonder if this will be confusing later on when the case manager asks the user about product price, if the number the user enters doesn't relate to single units? Though, this might not be something we can solve through validation.

Reply:
Yeah, the validation is really to help on the data-side of things. I mean, how big is a box?